include/swift/Runtime/GenericMetadataBuilder.h

```diff
@@ -0,0 +1,962 @@
+//===--- GenericMetadataBuilder.h - Build generic metadata. -----*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Builder for generic metadata, in-process and out-of-process.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_RUNTIME_GENERIC_METADATA_BUILDER_H
+#define SWIFT_RUNTIME_GENERIC_METADATA_BUILDER_H
+#include "swift/ABI/Metadata.h"
+#include "swift/Basic/MathUtils.h"
+#include "swift/Demangling/TypeLookupError.h"
+#include "llvm/Support/Casting.h"
+#include <dlfcn.h>
```

Comment:
`<dlfcn.h>` doesn't exist on Windows. We should probably conditionalise this.

```suggestion

#if SWIFT_STDLIB_HAS_DLADDR
#include <dlfcn.h>
#endif

```